File: MATLAB and R_Programs/forward.m
%% MATLAB script to show the position of transmitters and receivers 
% (Figure 1 in the report)
clear all;clc;
dx=1;%step along the horizontal
dy=1;%step along the vertical
x=0:dx:10;
y=0:dy:10;
nx=length(x)-1;ny=length(y)-1;% grid nx by ny
xmin=min(x);xmax=max(x);% end points of the horizontal
ymin=min(y);ymax=max(y);% end points of the vertical
rangex=(xmax-xmin);rangey=(ymax-ymin);
side1=5;     %Number of receivers along side 1
Rxy1=[(xmin+((1:side1)*rangex)/(side1+1))',ymax*ones(side1,1)];
side2=5;     %Number of receivers along side 2
Rxy2=[xmax*ones(side1,1),(ymin+((1:side2)*rangey)/(side2+1))'];
Rxy=[Rxy1;Rxy2];  %Receiver positions alongh side 1 and 2
side3=5;%Number of transmitters along side 3
Txy1=[xmin*ones(side3,1),(ymin+((1:side3)*rangey)/(side3+1))'];
side4=0;%No transmitters along side 4
Txy2=[(xmin+((1:side4)*rangex)/(side4+1))',ymin*ones(side4,1)];
Txy=[Txy1;Txy2];
b=1;          % initalizing number of rays
N_ray=(side1+side2)*(side3+side4);% number of rays from transmitter-receiver pair
ray=zeros(N_ray,4);% initializing ray matrix
for t=1:(side3+side4)
    for r=1:(side1+side2)
           ray(b,1)=Txy(t,1);% x components of the transmitters position
           ray(b,2)=Txy(t,2);% y components of the transmitters position
           ray(b,3)=Rxy(r,1);% x components of the receivers position
           ray(b,4)=Rxy(r,2);% y components of the receivers position
           b=b+1;
       end
end
%% Graphics
% grid plot
for j=1:length(y)
    plot(x,y(j)*ones(length(x)),'k','Linewidth',1)
    hold on
end
hold on
for k=1:length(x)
    plot(x(k)*ones(length(y)),y,'k','Linewidth',1)
    hold on
end
hold on
% Plot position of the transmitters and receivers
plot(Rxy(:,1),Rxy(:,2),'o','markersize',8,'MarkerFaceColor','r')
hold on
plot(Txy(:,1),Txy(:,2),'rs','markersize',8,'MarkerFaceColor','g')
hold on
% Plot ray path between transmitters and receivers
for i=1:N_ray
   h= plot([ray(i,1),ray(i,3)],[ray(i,2),ray(i,4)]);
    hold on
end
xlabel('Position (arbitrary units)')
ylabel('Position (arbitrary units)')
saveas(h,'fig1.png')

File: MATLAB and R_Programs/forward.R
## R script for Tomography
# by Fasil Tesema, November 28 2013, Dessie/Kombolcha, Ethiopia.
# Phd student at Bahir Dar University, Bahir Dar, Ethiopia
rm(list=(ls()))# clear all the variables
#setwd("C:/Users/Fasilo/Desktop/MATLAB and R_programs/")
nx=10;ny=10# grids in x and y direction (2D)
dx=1;dy=1# step of the grid
x=seq(0,nx,by=dx);y=seq(0,ny,by=dy) #nx=length(x)-1#ny=length(y)-1
xmin=min(x);xmax=max(x);ymin=min(y);ymax=max(y)
rangex=xmax-xmin;rangey=ymax-ymin
side1=5 #number of receivers on side 1 at the surface of the ground
Rxy1=matrix(c(t((xmin+(1:side1)*rangex)/(side1+1)),ymax*(t(rep(1,side1)))),nrow=side1,ncol=2)
side2=5 #number of receivers on side 2 (borehole 2)
Rxy2=matrix(c(xmax*(rep(1,side2)),t((ymin+(1:side2)*rangey)/(side2+1))),nrow=side2,ncol=2)
Rxy=rbind(Rxy1,Rxy2)
side3=5 #number of transmitters on side 3 (borehole 1)
Txy=matrix(c(xmin*(t(rep(1,side3))),t((ymin+(1:side3)*rangey)/(side3+1))),nrow=side3,ncol=2)
##########################################################################
# Graphics
plot(c(Txy[,1],Rxy[,1]),c(Txy[,2],Rxy[,2]),xlab="Position (arbitrary units)",
ylab="Position (arbitrary units)",ylim=c(0,10),pch=NA)
points(Txy[,1],Txy[,2],pch=15,col="green",cex=2)
points(Rxy[,1],Rxy[,2],pch=16,col="red",cex=3)
axis(1,at=0:10)
axis(2,at=0:10)
for (i in 1:length(Txy[,1])){for (j in 1:length(Rxy[,1])){
lines(c(Txy[i,1],Rxy[j,1]),c(Txy[i,2],Rxy[j,2]),type="l",col="blue")
}}
grid(lwd=1,lty="solid",col="black")
box(col="black",lwd=1)
